lib/restful_resource_bugsnag/middleware.rb

```diff
@@ -49,9 +49,15 @@ def request_host_from_exception(exception)
     end
 
     def attempt_json_parse(string)
-      JSON.parse(string)
+      redacted_string = redact_user_emails_from_errors(string)
+      JSON.parse(redacted_string)
     rescue JSON::ParserError, TypeError
-      string
+      redacted_string
+    end
+
+    def redact_user_emails_from_errors(string)
+      # Based on Devise.email_regexp
+      string&.gsub(/[^@\s]+@[^@\s]+/, '<email-address-redacted>')
```

Comment:
Given this is executed on the unparsed json I think we need to test that it does not screw up a valid json response.

I think the current regex is a bit too inclusive, so if you test it against:
`{msg:"Message with email: email@example.com"}`

the output string would be
`{msg:"Message with email: <email-address-redacted>`

the first part of the regex has also the same issue, if we have:
`{msg:"email@example.com is invalid"}`

I believe will return 
`<email-address-redacted> is invalid"}`

A way to fix it is by putting `"` in both character classes, e.g.
`string&.gsub(/[^@"\s]+@[^@"\s]+/, '<email-address-redacted>')`


I can't think of other edge cases though..

Reply:
@nexusventuri I updated the Regex and added some more specs for different structure messages. I also added another expectation in the shared example that checks that the response has actually been parsed.